src/test/java/uk/gov/hmcts/reform/sandl/snlevents/actions/hearing/UnlistHearingActionTest.java

```diff
@@ -159,15 +155,13 @@ public void generateFactMessages_shouldReturnMsgForUpdatedHearingParts() {
     public void act_shouldSetHearingPartSessionIdToNull() {
         action.getAndValidateEntities();
         action.act();
-
-        ArgumentCaptor<List<HearingPart>> captor = ArgumentCaptor.forClass((Class) List.class);
-
-        Mockito.verify(hearingPartRepository).save(captor.capture());
-        assertThat(captor.getValue().size()).isEqualTo(hearingVersions.size());
-        captor.getValue().forEach(hp -> {
-            assertNull(hp.getSessionId());
-            assertNull(hp.getSession());
-            assertNull(hp.getStart());
+        ArgumentCaptor<Hearing> captor = ArgumentCaptor.forClass(Hearing.class);
+        Mockito.verify(hearingRepository).save(captor.capture());
+        captor.getValue().getHearingParts().forEach(hp -> {
+            if (hp.getStatus().getStatus().equals(Status.Unlisted)) {
```

Comment:
in all the other cases shouldn't we assertNotNull?

Reply:
An unlisted hearing will only consist of hearingParts which are not assigned

Reply:
and you are sure in this test (and others similar) that all returned hp's will be of status unlisted? in such case, I would suggest removing the 'if' which will result in prooving in future that all hp's are unlisted and if it will fail, signal it.

Reply:
Yep you're right! - unlisting should only have one status; although the other action tests need the 'if' conditions to separate out the different statuses